src/content/reference/react/cache.md

```diff
@@ -0,0 +1,305 @@
+import {cache} from 'react';
+import {fetchTemperature} from './temperature';
+
+export default async function AnimatedWeatherCard({city}) {
+	const getTemperature = cache(fetchTemperature);
+	const temperature = await getTemperature(city);
+  // ...
+}
+
+// app.js
+import AnimatedWeatherCard from './animatedWeatherCard';
+import AnimatedWeatherCard from './animatedWeatherCard';
+
+function App() {
+	return (<>
+      <AnimatedWeatherCard city="Los Angeles" />
+      <MinimalWeatherCard city="Los Angeles" />
+	</>);
+}
+```
+
+In this example, `MinimalWeatherCard` and `AnimatedWeatherCard` are not leveraging the same cache and so `fetchTemperature` is double invoked with the same argument `"Los Angeles"`.
+
+This is because each call to `cache` creates a new memoizing function that uses a different cache for look-up. For `AnimatedWeatherCard` and `MinimalWeatherCard` to use the same cache, `getTemperature` should be defined in a mutually accessible scope, like moving the definition to `temperature.js`.
+
+In addition, `AnimatedWeatherCard` is calling `cache` in its component render. This means that each time `AnimatedWeatherCard` is rendered, a new memoizing function is created so there will never be an opportunity for cache-sharing.
+
+</Pitfall>
+
+### Preload data {/*preload-data*/}
+
+By wrapping a long-running data fetch in `cache`, you can kick off the asynchronous work prior to rendering the component.
+
+```jsx
+const getUser = cache(async (id) => {
+  return await db.user.query(id);
+}
+
+async function Profile({id}) {
+  const user = await getUser(id);
+  // ...
+}
+
+function Page({id}) {
+  getUser(id);
+  // ... some computational work
+  return (
+    <>
+	    ...
+      <Profile id={id} />
+    </>
+  );
+}
+```
+
+When rendering `Page`, the component calls `getUser` but note that it doesn't use the returned data. This early `getUser` call kicks off the asynchronous database query that occurs while `Page` is doing other computational work and rendering children. 
+
+When rendering `Profile`, we call `getUser` again. In the ideal case, the initial `getUser` call has already returned and cached the user data. So when `Profile` asks and waits for this data, it can simply read from the cache without requiring another remote procedure call. Even if the data request hasn't been completed, preloading data in this pattern reduces delay in data-fetching..
+
+<Pitfall>
+
+Memoization only occurs within component renders.
+
+```jsx
+import {cache} from 'react';
+
+const getUser = cache(async (userId) => {
+  return await db.user.query(userId);    
+});
+
+// This call to `getUser` will not update the cache.
+getUser('demo-id');
+
+async function DemoProfile() {
+  // This `getUser` will have a cache miss
+  const user = await getUser('demo-id');
+  return (<Profile user={user} />);
+}
+```
+
+You may attempt to preload some work outside of a component. React only provides cache access to the memoizing function in a component render. 
```

Comment:
I think I may have missed why this is a limitation, does anyone know?

Reply:
The [cache uses context](https://github.com/facebook/react/blob/f101c2d0d3a6cb5a788a3d91faef48462e45f515/packages/react-reconciler/src/ReactFiberCache.js#L29)